### ISSUE TITLE
nothing on search results page for "selinux-policy" unless entered in full

### ISSUE THREAD

**Issue by oishommau** [fedora-infra/bodhi#870]
entering "selinux" produces no results (https://bodhi.fedoraproject.org/updates/?packages=selinux)

entering up to "selinux-polic" still produces no results (https://bodhi.fedoraproject.org/updates/?packages=selinux-polic)

you can only get results for selinux-policy if you type the whole thing in (https://bodhi.fedoraproject.org/updates/?packages=selinux-policy)

note that it does however show up in the suggestions dropdown for any of the above cases

**Comment by oishommau** [comment]
originally mentioned in #692 (but apparently unrelated to that bug, since not addressed by the fix) -- it's also not possible to find selinux-policy-targeted at all

looking at this again now, it seems like selinux-policy-targeted is a phantom package tied to the selinux-policy package? can anyone explain what's going on here?

should this be a separate bug, or is it related to the issue originally described?

if it's some kind of "alias", perhaps bodhi should handle it accordingly.

**Comment by bowlofeggs** [comment]
Hello @oishommau!

I think what you might be experiencing is the difference between what RPM would think of as a "source package" and a "subpackage". A source package has a 1:1 relationship with a spec file. If you look for selinux-policy in [pkgdb](https://admin.fedoraproject.org/pkgdb/package/rpms/selinux-policy/), you can see that pkgdb is like Bodhi in that it knows about source packages.

However, a single spec file can generate more than one RPM that end users can install. And if you look in Koji at a [build](https://koji.fedoraproject.org/koji/buildinfo?buildID=919436) from selinux-policy, you can see that it results in several RPM packages for end-users to interact with, including selinux-policy-targeted.

Unfortunately, Bodhi does not currently know about subpackages and concerns itself only with source packages. This is why the search doesn't find what you are looking for. I could see adding a feature to give Bodhi the ability to find source packages based on subpackage name searches, but I would consider that more of an RFE than a bug.

In the meantime, it is possible to ask dnf to tell you the source package than generated an RPM:

```
$ dnf repoquery -s selinux-policy-targeted
selinux-policy-3.13.1-190.fc24.src.rpm
selinux-policy-3.13.1-191.24.fc24.src.rpm
```

Sorry for the confusion, but I hope that dnf trick helps!

**Comment by oishommau** [comment]
thanks for the explanation, however, that still doesn't seem to explain the original issue described -- that a search for "selinux" doesn't produce any results at all. should i split one of these issues into a separate issue?

**Comment by bowlofeggs** [comment]
Hello @oishommau!

My comment was in response to your second comment and not really in response to the original report. I agree that it would be ideal if the search box used an ```ilike``` query instead of an exact match query. If you want the search to know about subpackages, feel free to file a new RFE but note that it would not be trivial to add that to Bodhi since Bodhi currently has no knowledge of subpackages.

**Comment by bowlofeggs** [comment]
The patch for this issue should be included in the upcoming 2.10.0 release:

https://github.com/fedora-infra/bodhi/pull/1744
